setup.py

```diff
@@ -683,6 +725,24 @@ def get_snappy_clib(field=None):
     return config[field]
 
 
+def get_sperr_clib(field=None):
+    """sperr static lib build config"""
+    sperr_dir = "src/SPERR"
+
+    config = dict(
+        sources=glob(f"{sperr_dir}/src/*.cpp"),
+        include_dirs=[f"{sperr_dir}/include"],
+        macros=[
+            ("SPERR_VERSION_MAJOR", 0),
```

Comment:
@shaomeng I had to set this macro which is used in the code.
Is there a better way?

Reply:
`USE_VANILLA_CONFIG` is supposed to be used like this, so there's no problem.
`SPERR_VERSION_MAJOR` is populated by cmake from this line:
https://github.com/NCAR/SPERR/blob/ef135ee9d87fc9b64cdd213e93e03fedf151e7f5/CMakeLists.txt#L5
So I think if it's possible to extract that first digit (`0` right now) after the `VERSION` keyword, it'd be future proof. Otherwise, we probably need to manually update the `setup.py` once SPERR gets a version bump. I don't think it's gonna happen very often though.

Reply:
OK, let's keep it manual for now